### PR TITLE
fix(server): Remove mutex around the storage adapter

### PR DESCRIPTION
**Pull request by jchauncey**
It was once possible to change the storage adapter of logger mid flight so there was a mutex around that ability. However, now that is not possible. So we should just remove that code. This also helps fix a problem where log entires are never stored in the ring buffer

Test Steps:

1. Install workflow-dev as that is the easiest way to get the logger system up
1. Build and replace the logger pod with the new hawtness in this PR (using the make targets provided)
1. Create an application that spits out stuff to stdout like the awesome `example-go` app
1. Generate logs somehow (like curling the app that you just deployed)
1. Run `deis logs` and make sure your new hawt logger sends data back

### CODE REVIEW

**Comment by arschles** [comment]
@jchauncey just tested manually, and things seem to work, using an app deployed with `deis pull deis/example-go`:

```console
ENG000656:logger aaronschlesinger$ deis logs -a gotest
gotest-v2-cmd-jagwr -- 2016/04/12 22:10:30 10.168.1.5:47374 GET /
gotest-v2-cmd-jagwr -- 2016/04/12 22:10:33 10.168.1.5:47380 GET /
gotest-v2-cmd-jagwr -- 2016/04/12 22:10:37 10.168.1.5:47389 GET /
gotest-v2-cmd-jagwr -- 2016/04/12 22:10:41 10.168.1.5:47401 GET /
gotest-v2-cmd-jagwr -- 2016/04/12 22:10:43 10.168.1.5:47405 GET /
gotest-v2-cmd-jagwr -- 2016/04/12 22:10:45 10.168.1.5:47415 GET /
```

File: syslogish/server.go [hunk 1]
```diff
@@ -116,13 +110,6 @@ func (s *Server) processStorage() {
...
 				// We only want to store deis app log messages
 				if labels != nil && labels["app"] != nil && labels["heritage"] != nil && labels["heritage"].(string) == "deis" {
-					// Get a read lock to ensure the storage adapater pointer can't be nilled by the
-					// configurer in the time between we check if it's nil and the time we invoke .Write()
-					// upon it.
-					s.adapterMutex.RLock()
-					// DONT'T defer unlocking... defered statements are executed when the function returns,
-					// but we are inside an infinite loop here.  If we defer, we would never release the
-					// lock.  Instead, release it manually below.
 					if s.storageAdapter != nil {
```
Comment (arschles) [inline syslogish/server.go]: This is the only place I can see that checks any shared state. Do we need to check if the adapter is `nil` at all?

Comment (arschles) [inline syslogish/server.go]: Same with the below checks

**Comment by mboersma** [comment]
I tested this locally on kube-solo with example-clojure-ring:
```console
$ deis logs
verbal-gadabout-v2-web-avjsd -- Retrieving org/clojure/tools.nrepl/0.2.12/tools.nrepl-0.2.12.pom from central
verbal-gadabout-v2-web-avjsd -- Retrieving clojure-complete/clojure-complete/0.2.4/clojure-complete-0.2.4.pom from clojars
verbal-gadabout-v2-web-avjsd -- Retrieving org/clojure/tools.nrepl/0.2.12/tools.nrepl-0.2.12.jar from central
verbal-gadabout-v2-web-avjsd -- Retrieving clojure-complete/clojure-complete/0.2.4/clojure-complete-0.2.4.jar from clojars
verbal-gadabout-v2-web-avjsd -- Picked up JAVA_TOOL_OPTIONS: -Xmx350m -Xss512k -Dfile.encoding=UTF-8
verbal-gadabout-v2-web-avjsd -- 2016-04-12 22:34:28.827:INFO::main: Logging initialized @5244ms
verbal-gadabout-v2-web-avjsd -- 2016-04-12 22:34:30.166:INFO:oejs.Server:main: jetty-9.2.10.v20150310
verbal-gadabout-v2-web-avjsd -- 2016-04-12 22:34:30.268:INFO:oejs.ServerConnector:main: Started ServerConnector@433f1a20{HTTP/1.1}{0.0.0.0:5000}
verbal-gadabout-v2-web-avjsd -- 2016-04-12 22:34:30.273:INFO:oejs.Server:main: Started @6686ms
```
:+1:

**Pull request merged**